gtk4/src/subclass/adjustment.rs

```diff
@@ -0,0 +1,73 @@
+use gtk_sys;
+
+use glib::subclass::object::ObjectImpl;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, Object};
+
+use Adjustment;
+
+pub trait AdjustmentImpl: AdjustmentImplExt + ObjectImpl {
+    fn changed(&self, adjustment: &Self::Type) {
+        self.parent_changed(adjustment)
+    }
+
+    fn value_changed(&self, adjustment: &Self::Type) {
+        self.parent_value_changed(adjustment)
+    }
+}
+
+pub trait AdjustmentImplExt: ObjectSubclass {
+    fn parent_changed(&self, adjustment: &Self::Type);
+    fn parent_value_changed(&self, adjustment: &Self::Type);
+}
+
+impl<T: AdjustmentImpl> AdjustmentImplExt for T {
+    fn parent_changed(&self, adjustment: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkAdjustmentClass;
+            let f = (*parent_class)
+                .changed
+                .expect("No parent class impl for \"changed\"");
```

Comment:
This and the other vfunc is optional and simply nothing should happen if it's `NULL`. See the C code.